hooks/post-checkout

```diff
@@ -1,112 +1,40 @@
-            validate_version_file
-            UNRESOLVED_VERSION=$(cat ${INPUT_VERSION_FILE:?} | tr -s ' ')
-        else # `version_file` was not provided
-            echo "ERROR: neither \`version\` nor \`version_file\` were provided."
-            exit 1
-        fi
-    fi
-}
+# The external repository that we will clone.
+# Example value: https://github.com/JuliaCI/julia-buildkite
+INPUT_REPO_URL="$(get_meta_env_value REPO_URL)"
 
-function get_unresolved_version_subsequent_jobs() {
-    # In subsequent jobs, the `version` input must be provided.
-    if [[ -z "${INPUT_VERSION?}" ]]; then
-        echo "ERROR: This is not the first job, so the \`version\` input must be provided."
-        exit 1
-    fi
-    # In subsequent jobs, the `version_input` must not be provided.
-    if [[ ! -z "${INPUT_VERSION_FILE?}" ]]; then
-        echo "ERROR: This is not the first job, so the \`version_file\` input must not be provided."
-        exit 1
-    fi
-    UNRESOLVED_VERSION="${INPUT_VERSION:?}"
-}
+# Folder the external repository will be cloned into, defaults to `.buildkite`
+INPUT_FOLDER="$(get_meta_env_value FOLDER .buildkite)"
 
-function get_unresolved_version() {
-    if [[ "${INPUT_THIS_IS_FIRST_JOB?}" == "true" ]]; then
-        get_unresolved_version_first_job
-    else
-        get_unresolved_version_subsequent_jobs
-    fi
-}
+# Version of the external repository that we're going to clone
+# This can be a gitsha, a gitref, or a file containing one of the two
+INPUT_VERSION="$(get_meta_env_value VERSION)"
 
-### End helper functions
 
-### Begin the main part of the script
+# Clone external repo to the local path:
+git clone "${INPUT_REPO_URL}" "${INPUT_FOLDER}"
 
-INPUT_EXTERNAL_REPO="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_EXTERNAL_REPO:?}"
-INPUT_FOLDER="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER:?}"
+# Resolve the version, now that we have all the files locally:
+RESOLVED_VERSION="$(resolve_version "${INPUT_VERSION}" "${INPUT_FOLDER}")"
 
-INPUT_THIS_IS_FIRST_JOB="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_THIS_IS_FIRST_JOB-}"
-INPUT_VERSION="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION-}"
-INPUT_VERSION_FILE="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE-}"
+# Check out the resolved version
+git -C "${INPUT_FOLDER}" checkout "${RESOLVED_VERSION}"
 
-echo "--- Print the plugin inputs"
-get_unresolved_version
-echo "INFO: The Buildkite config version (unresolved) is: ${UNRESOLVED_VERSION:?}"
-echo "INFO: The Buildkite config files will be downloaded from the following repository: ${INPUT_EXTERNAL_REPO:?}"
-echo "INFO: The Buildkite config files will be downloaded into the following folder: ${INPUT_FOLDER:?}"
+# Report the values via environment variables
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="${RESOLVED_VERSION}"
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="${INPUT_REPO_URL}"
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${INPUT_FOLDER}"
```

Comment:
Since we will set these as metadata, I don't think we also need to export them as environment variables.

```suggestion
```

Reply:
I dunno, I kind of like having the resolved version and such immediately available in the environment, I think it's still useful.  Especially since if the user defines values in the YAML, they'll probably expect for the environment variables to be used since that's how they're communicated to the plugin from the YAML.  So I think it's best to just remain consistent across all methods of communication.